packages/fxa-auth-server/lib/account-delete.ts

```diff
@@ -181,60 +203,93 @@ export class AccountDeleteManager {
    * Delete the account from the FxA database, OAuth database, pushbox database, and
    * cancel any active subscriptions.
    *
-   * @param uid string
-   * @param options AccountDeleteOptions optional object with an optional `notify` function that will be called after the account's removed from MySQL.
+   * @param uid User id
+   * @param reason Enum describing the reason for the deletion.
+   * @param customerId Stripe customer id if the user had/has a subscription.
    */
   public async deleteAccount(
     uid: string,
-    customerId?: string,
-    options?: AccountDeleteOptions
+    reason: ReasonForDeletion,
+    customerId?: string
   ) {
-    await this.deleteAccountFromDb(uid, options);
+    await this.deleteAccountFromDb(uid);
     await this.deleteOAuthTokens(uid);
     // see comment in the function on why we are not awaiting
     this.deletePushboxRecords(uid);
 
-    await this.deleteSubscriptions(uid, customerId);
+    await this.deleteSubscriptions(uid, reason, customerId);
     await this.deleteFirestoreCustomer(uid);
   }
 
-  public async cleanupAccount(uid: string, customerId?: string) {
-    await this.deleteSubscriptions(uid, customerId);
-    await this.deleteFirestoreCustomer(uid);
-    await this.deleteOAuthTokens(uid);
+  /**
+   * Delete the account from the FxA database, OAuth database, and pushbox database.
+   * This method is intended to be used when the user is waiting to be deleted from
+   * the system so it only clears up the immediate database records and logs out the
+   * user. The rest of the cleanup is handled later after queuing the account for
+   * deletion.
+   */
+  public async quickDelete(uid: string, reason: ReasonForDeletion) {
+    if (reason !== ReasonForDeletionOptions.UserRequested) {
+      throw new Error('quickDelete only supports user requested deletions');
+    }
+
+    try {
+      await this.deleteAccountFromDb(uid);
+      await this.deleteOAuthTokens(uid);
+    } catch (error) {
+      // If the account wasn't fully deleted, we should log the error and
+      // still queue the account for cleanup.
+      this.log.error('quickDelete', { uid, error });
```

Comment:
Yea, I don't think that's any worse/better than it is now, because their db data is possibly in a corrupted format now. We might've deleted critical records needed to login before the failure happened, which may prevent it from going any farther. Catching and continuing to enqueue here is specifically to ensure we can keep retrying to fully clean it up.